### PR TITLE
Add dockerized version of Protect

### PR DESCRIPTION
**Pull request by cket**

### CODE REVIEW

**Comment by cket** [comment]
@arkal this is ready for review whenever you're free

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Install Toil
+RUN pip install toil[aws]==3.3.1
+
+# Install s3am
+RUN pip install s3am
+
+# Install ProTECT
+RUN pip install protect==2.3.0
+
+# s3am requires an older version of bd2k-python-lib than ProTECT requires.
```
Comment (arkal) [inline docker/Dockerfile]: This is not true, right? Isn't this a Toil issue?

**Comment by cket** [comment]
@arkal we should also change jenkins so it builds & pushes the docker containers to quay - I took a look at jenkins.sh, it seems like we can make the changes directly in the jenkins build config, right?

**Comment by arkal** [comment]
Done.
Jenkins, test this please.

**Comment by cket** [comment]
@arkal any idea whats going on with the test?

**Comment by arkal** [comment]
Jenkins, test this please.

File: docker/Makefile.template [hunk 2]
```diff
@@ -14,7 +14,6 @@ build: Dockerfile
 push: build
 	# Requires ~/.dockercfg
 	docker push ${name}:${tag}
-	docker push ${name}:latest
```
Comment (arkal) [inline docker/Makefile.template]: why? Shouldn't we tag ${tag} as `latest` and push `latest`as well?

Comment (cket) [inline docker/Makefile.template]: It's down to your preference - I'm really just copying the Toil appliance build methodology.

Comment (arkal) [inline docker/Makefile.template]: Then can we have a latest, please

**Comment by arkal** [comment]
This one currently builds protect==2.3.0. A method to create a new docker for every release of protect is tracked in #142

**Pull request merged**